lib/query_components/base_component.rb

```diff
@@ -2,14 +2,13 @@ module QueryComponents
   class BaseComponent
     include Elasticsearch::Escaping
 
-    attr_reader :search_params
+    attr_reader :search_params, :search_term
 
     def initialize(search_params = SearchParameters.new)
       @search_params = search_params
+      @search_term   = @search_params.query.strip.sub(/^"/, "").sub(/"$/, "") unless @search_params.query.nil?
```

Comment:
Is this just here to abstract shared code between `core_query` and `text_query`?  If so, I won't bother abstracting it (ie, I'd just have the same code twice), because we're working towards removing `core_query`, and that's actually easier if it's more self-contained.

Also, I'm uncomfortable about the `unless` clause here - if I read it right, this will mean that `@search_term` isn't even defined if `search_params.query` is nil, which seems likely to result in unexpected errors due to failing to look up the variable.  I'd rather `@search_term` was at least defined as being `nil` in that case.